src/lib/LLVM/V9/JIT.hs

```diff
@@ -141,12 +154,21 @@ unloadNativeModule NativeModule{..} = do
   LLVM.disposeModule llvmModule
   LLVM.disposeContext llvmContext
 
-withNativeModule :: JIT -> LLVM.AST.Module -> CompilationPipeline -> (NativeModule -> IO a) -> IO a
-withNativeModule jit m p = bracket (compileModule jit m p) unloadNativeModule
+withNativeModule :: JIT -> [ObjectFileContents] -> LLVM.AST.Module -> CompilationPipeline -> (NativeModule -> IO a) -> IO a
+withNativeModule jit objs m p = bracket (compileModule jit objs m p) unloadNativeModule
 
 getFunctionPtr :: NativeModule -> String -> IO (FunPtr a)
 getFunctionPtr NativeModule{..} funcName = do
   let JIT{..} = moduleJIT
   symbol <- OrcJIT.mangleSymbol compileLayer $ fromString funcName
   Right (OrcJIT.JITSymbol funcAddr _) <- OrcJIT.findSymbolIn compileLayer moduleKey symbol False
   return $ castPtrToFunPtr $ wordPtrToPtr funcAddr
+
+loadObjectFile :: JIT -> IO OrcJIT.ModuleKey -> ObjectFileContents -> IO ()
+loadObjectFile jit makeKey objFileContents = do
+  withSystemTempFile "objfile.o" \path h -> do
+    BS.hPut h objFileContents
+    hFlush h
+    key <- makeKey
```

Comment:
Is there any good reason why this has to make the key here, instead of accepting it as an argument?